mlflow/pipelines/steps/train.py

```diff
@@ -343,6 +343,7 @@ def my_warn(*args, **kwargs):
                         evaluator_config={
                             "log_model_explainability": False,
                             "pos_label": self.positive_class,
+                            "metric_prefix": f"{dataset_name}_",
```

Comment:
This change affects `validation_criteria` in the pipeline YAML. They are currently NOT prefixed.

```yaml
    validation_criteria:
      - metric: root_mean_squared_error
        threshold: 10
      - metric: mean_absolute_error
        threshold: 50
      - metric: weighted_mean_squared_error
        threshold: 20
```

Reply:
We should process the metric names internally. Users should not have to prefix them

Reply:
Do we want to allow users to use prefixed metrics (e.g. `test_rmse`)?


A potential use case is:

```yaml
    validation_criteria:
      # If training_xxx_metric is lower than y,
      # it's possible that the mode is underfitted.
      - metric: training_xxx_metric
        threshold: y
```

Reply:
The validation criteria is always assumed to be defined on the test dataset, so I think assuming test is the best thing to do here.

Reply:
Sounds good. I'll make the change to process <metric_name> as validation_<metric_name> internally.

Reply:
thanks so much, @harupy ! :)